### ISSUE TITLE
System Notification Breaks Experience Fragment based Target Activities

### ISSUE THREAD

**Issue by ulhep** [Adobe-Consulting-Services/acs-aem-commons#3362]
## Required Information

* [ ] AEM Version, including Service Packs, Cumulative Fix Packs, etc: AEM 6.5.20.0
* [ ] ACS AEM Commons Version: 5.0.14
* [ ] Reproducible on Latest? Not tried

### Expected Behavior

System Notifications should display to Content Authors.

### Actual Behavior

The notifications display just fine. However, when you export or update an Experience Fragment to Target, System Notification code is adding document.write() to the HTML. This BREAKS Adobe Target Activities using those XFs and causes widespread pandemonium across an organization.

### Steps to Reproduce

Using above versions of AEM and ACS Commons, create and enable a System Notification. Then, export / update an XF to Target and observe the HTML.

### Links

On click of "Export to Target" or "Update to Target", to start with, it initiates a following request:

https://author.uat.apps.xxxx.com/content/experience-fragments/www-xxxx-touch/nl/sandbox/jan/banner/master.html?wcmmode=disabled

The HTML markup received as response contains following system notification code:

<script>if(window === top) {   window.jQuery || document.write('<script src="/etc/clientlibs/granite/jquery.js"><\/script>');   document.write('<script src="/apps/acs-commons/components/utilities/system-notifications/notification/clientlibs.js"><\/script>');}</script>
</body>

And. this causes the issue while using the exported exp fragments in the Adobe Target editor

**Comment by ulhep** [comment]
To fix the issue, we can add following change in AbstractHtmlRequestInjector.java
``
protected boolean accepts(final ServletRequest servletRequest,
                            final ServletResponse servletResponse) {

       . . . 
       . . .

        // Do not apply when exporting Target Offers
        if (request instanceof SlingHttpServletRequest) {
            final SlingHttpServletRequest slingRequest = (SlingHttpServletRequest) request;
            . . . 
            . . .
            if (StringUtils.contains(slingRequest.getQueryString(), "wcmmode=disabled")) {
                return false;
            }
        }

        . . .
        return true;
    }
```

**Comment by ulhep** [comment]
Trying to push a bugfix for it, but don't have commit rights
```
remote: Permission to Adobe-Consulting-Services/acs-aem-commons.git denied to ulhep.
fatal: unable to access 'https://github.com/Adobe-Consulting-Services/acs-aem-commons.git/': The requested URL returned error: 403
```

**Comment by ulhep** [comment]
finally could raise a pull request:
https://github.com/Adobe-Consulting-Services/acs-aem-commons/pull/3363

**Comment by ulhep** [comment]
@davidjgonzalez 
When will we create a new release?
Currently, acs-aem-commons-6.6.0 is the latest created on 15th April 2024

**Comment by davidjgonzalez** [comment]
Fixed in 6.6.2 which is now released
